src/kili/services/plugins/upload.py

```diff
@@ -186,11 +212,16 @@ def _retrieve_requirements(self) -> Union[Path, None]:
         return file_path
 
     @staticmethod
-    def _parse_script(file_path: Path):
+    def _parse_script(script_path: Path, name: str):
         """
-        Method to detect indentation errors in the script
+        Method to detect indentation and class errors in the script
         """
-        with file_path.open("r", encoding="utf-8") as file:
+        if str(script_path) == "main.py" and not check_file_contains_handler(
```

Comment:
If this is a Path, couldn't it contain more elements ? I mean for example `path/to/folder/main.py` so your check on main would not work ?

Reply:
Yes, that's right. @hguerni did you test your feature before pushing your branch?

Reply:
Yes I did but only with file in root, now done using path.name